### ISSUE TITLE
Add sync feature

### ISSUE THREAD

**Issue by wlthomson** [openmsupply/remote-server#385]
@craigdrown noted that it would be good to have a way to disable sync. This got me thinking about the idea of putting sync behind a cfg flag and potentially putting it in a separate crate.

The problem with the crate approach is that sync currently depends on repositories and diesel mappings from the remote-server crate. In the future, it would be a potential idea to make sync a stand-alone module that can take some configuration parameters, including a database handle and record mappings, and decouple it entirely from the core `remote-server` crate.

For now, I think just putting sync functionalty behind a conditional compilation feature will do the job?

**Comment by craigdrown** [comment]
@wlthomson  thanks - this will be a needed feature before release anyhow (at least stopping processing of a sync queue from a backup database)

**Comment by wlthomson** [comment]
Good point. I think the best way to do that might be to expose a way to send messages to the actors in the same way they are sent internally... that would be pretty cool use-case for something like gRPC and would likely have a lot of use-cases for plugins too! 

For now, I will think over some options for a simpler interface, maybe adding something using the existing HTTP API.
